Contest_API.md

```diff
@@ -142,7 +142,13 @@ their expected behavior, if implemented.
 
 Standard [HTTP status
 codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) are
-returned to indicate success or failure.
+returned to indicate success or failure. When there is a failure (status codes 4xx or 5xx), the
+response message body must include a JSON element that contains the attributes 'code' (a number, identical to the HTTP status code returned) and 'message' (a string) with further information suitable for the user making the request, as per the following example:
```

Comment:
There will (maybe?) not always be a good error message. I think it would be better for `message` to be optional.

This kind of setup is not uncommon. Do we know how other (good) APIs handle this?

Reply:
DOMjudge uses a package for its API and that uses exactly tje way Tim suggested, so it seems this way is at least used more

Reply:
@nickygerritsen What package are you using?

Is it also treating `message` as a mandatory field?

Reply:
I think it is in the default Symfony (a php framework) code but I will double check when on a bigger screen.

I think it always exposes a message, since it uses exceptions to generate them and exceptions have a message

Reply:
FWIW, I looked at a few REST APIs for background, most were either missing error handling or overly complicated. I did find this used in two other places. I don't remember one but the other is an OpenAPI/Swagger example: https://swagger.io/docs/specification/describing-responses/ (note that code is a string, maybe that's where I got it from :) ).

IMHO the message should be mandatory so that the client always has something it can show the user along with the code. The server can keep it generic/error code lookup if necessary, but in my experience there's almost always something more useful to say.

Reply:
FYI the library we use and that does this by default is FOSRestBundle, where it has [this logic](https://github.com/FriendsOfSymfony/FOSRestBundle/blob/3.x/Serializer/Normalizer/FlattenExceptionHandler.php#L120-L121).
It always has a message and the code is an integer.